Source/EasyNetQ/RabbitAdvancedBus.cs

```diff
@@ -427,38 +428,40 @@ public Task<IQueue> QueueDeclareAsync()
                 return new Queue(name, exclusive);
             }
 
-            var arguments = new Dictionary<string, object>();
+            if (arguments == null)
+                arguments = new Dictionary<string, object>();
+
             if (perQueueMessageTtl.HasValue)
             {
-                arguments.Add("x-message-ttl", perQueueMessageTtl.Value);
+                arguments["x-message-ttl"] = perQueueMessageTtl.Value;
```

Comment:
So this is **additional** arguments only and all known arguments will be overriden by named arguments from method signature. It is worth reflecting in the comments to avoid misunderstandings.

Reply:
Yes, logic behind this behavior is `if you specify some arguments in this collection - then you certainly know what you're doing and all the defaults must be overridden with these explicitly supplied values`

Reply:
Yes, I see that it did not get worse.